selftests/unit/test_job.py

```diff
@@ -39,7 +39,33 @@ def _find_simple_test_candidates(candidates=None):
     def test_job_empty_suite(self):
         args = argparse.Namespace(base_logdir=self.tmpdir)
         self.job = job.Job(args)
-        self.assertIsNone(self.job.test_suite)
+        # Job without setup called
+        none_attrs = ("logdir", "replay_sourcejob", "result", "sysinfo",
+                      "test_runner", "test_suite", "tmpdir")
+        for attr in none_attrs:
+            self.assertIsNone(getattr(self.job, attr), "job.%s is not None "
+                              "(%s)" % (attr, getattr(self.job, attr)))
+        self.assertEquals(self.job.args, args)
+        self.assertEquals(self.job.exitcode, exit_codes.AVOCADO_ALL_OK)
+        self.assertEquals(self.job.references, [])
+        self.assertEquals(self.job.status, "RUNNING")
+        uid = self.job.unique_id
+
+        # Job with setup called
+        self.job.setup()
+        self.assertTrue(self.job.logdir)
```

Comment:
Even though `None` will evaluate to `False`, `assertIsNotNone` is better suited here.  That actually matches the suggestion about using plain `assertIsNone` before `setup()` is called.

Reply:
I'm not sure which `None` you have in mind :-/